integration/basic_test.go

```diff
@@ -635,3 +635,28 @@ func TestPingCommand(t *testing.T) {
 		require.Empty(t, actualDatabases.Databases)
 	})
 }
+
+func TestMutatingClientMetadata(t *testing.T) {
```

Comment:
I see that the example in the original issue contains `client` field in `runCommand` call. However, in the test I don't see `client` field being set. I believe we should add this test case to ensure the problem is fixed.